src/accelerate/data_loader.py

```diff
@@ -692,6 +699,7 @@ def prepare_data_loader(
             new_dataset,
             device=device if put_on_device and state.distributed_type != DistributedType.TPU else None,
             sampler=new_batch_sampler,
+            batch_size=None,
```

Comment:
Would reuse the same `batch_size` as for the original `datalaoder` here.

Reply:
This is necessary else batch_sampler would get created with batch_size=1 when it shouldn't do that as the sampler is already an instance of batch_sampler. This line here: https://github.com/pytorch/pytorch/blob/master/torch/utils/data/dataloader.py#L357

This leads to the issue <a> specified in the PR description

Reply:
But it's not our place to fix users mistakes and they might have wanted to use `batch_size=1` with this for some reason instead of `batch_size=None`.

Reply:
Here, the Dataloader default is batch_size=1 and not the one fixed by user. The one fixed by user is used which is part of sampler which is an instance of BatchSampler. When sampler is an object of BatchSampler, batch_sampler shouldn't be created and for that batch_size to dataloader has to be None so that dataloader can use sampler directly to create batches.

Reply:
Yes, `batch_size` should be None, but in the examples of the issue linked above, the user did not do that. We should honor the same argument as was passed to the original dataloader which is saved [here](https://github.com/pytorch/pytorch/blob/d2d145a40001d1e1f815a144160bd0b8d0f60ea0/torch/utils/data/dataloader.py#L361).

Reply:
Got it. However, even if the user passes `None` to dataloader, it isn't being passed to the final `DataLoaderShard` as batch_size is part of `ignore_kwargs` and that leads to default value of 1 being used. https://github.com/huggingface/accelerate/blob/main/src/accelerate/data_loader.py#L659

Reply:
Done.